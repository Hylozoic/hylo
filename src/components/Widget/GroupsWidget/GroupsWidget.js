import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import Slider from 'react-slick'
import { DEFAULT_BANNER, DEFAULT_AVATAR } from 'store/models/Group'
import { createGroupUrl, groupUrl, groupDetailUrl } from 'util/navigation'

import 'slick-carousel/slick/slick.css'
import 'slick-carousel/slick/slick-theme.css'
import './GroupsWidget.scss'

const { array, object } = PropTypes

const sliderSettings = {
  dots: true,
  infinite: false,
  speed: 500,
  slidesToShow: 2.1,
  arrows: true,
  slidesToScroll: 1,
  responsive: [
    {
      breakpoint: 600,
      settings: {
        slidesToShow: 1.1,
        slidesToScroll: 1
      }
    }
  ]
}

export default class GroupsWidget extends Component {
  static propTypes = {
    group: object,
    items: array
  }

  render () {
    const { items = [], routeParams } = this.props

    return (
      <div styleName='groups'>
        <Slider {...sliderSettings}>
          {items && items.map(g => <div styleName='group' key={g.id}>
            <div>
              <div styleName='content'>
                <div styleName='group-avatar'><img src={g.avatarUrl || DEFAULT_AVATAR} /></div>
                <div styleName='group-name'>{g.name}</div>
                <div styleName='member-count'>{g.memberCount} member{g.memberCount !== 1 ? 's' : ''}</div>
                <div styleName='group-description'>
                  {g.description}
                  {g.description && g.description.length > 140 && <div styleName='descriptionFade' />}
                </div>
                {g.memberStatus === 'member'
                  ? <div styleName='is-member'><Link to={groupUrl(g.slug)}><span>Member</span><span styleName='visit'>Visit</span></Link></div>
                  : g.memberStatus === 'requested'
                    ? <div styleName='isnt-member'><Link to={groupDetailUrl(g.slug, routeParams)}><span>Pending</span><span styleName='visit'>View</span></Link></div>
                    : <div styleName='isnt-member'><Link to={groupDetailUrl(g.slug, routeParams)}><span>View</span><span styleName='visit'>View</span></Link></div>
                }
              </div>
            </div>
            <div styleName='background' style={{ backgroundImage: `url(${g.bannerUrl || DEFAULT_BANNER})` }} ><div styleName='fade' /></div>
          </div>)}
          <div styleName='createGroup'>
            <div>
              <Link to={createGroupUrl(routeParams)}>+ Create Group</Link>
              {/* might have to make this conditional, to suit a wider range of uses */}
            </div>
          </div>
        </Slider>
        <div styleName='groupBumper' />
      </div>
    )
  }
}
