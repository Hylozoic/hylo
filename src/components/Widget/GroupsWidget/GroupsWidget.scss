.groups {
  padding: 0 15px;

  .group {
    height: 320px;
    position: relative;
    padding: 15px;
    text-align: center;
    box-shadow: 0px 5px 10px rgba(0, 0, 0, 0.15);

    > div {
      height: 100%;
      display: flex;
      align-items: center;
      justify-content: center;
    }

    .content {
      position: relative;
      z-index: 10;

      .group-avatar {
        text-align: center;

        img {
          margin: 0 auto 15px auto;
          width: 100px;
          height: 100px;
          border-radius: 10px;
          box-shadow: 0px 5px 10px rgba(0, 0, 0, 0.15);
        }
      }

      .group-name {
        font-size: 16px;
        font-weight: bold;
        color: rgba(255, 255, 255, 1);
      }

      .member-count {
        font-size: 12px;
        color: rgba(255, 255, 255, .6);
        padding-bottom: 10px;
      }

      .group-description {
        font-size: 12px;
        color: rgba(255, 255, 255, 1);
        padding-bottom: 10px;
      }

      .is-member a, .isnt-member a {
        color: rgba(255, 255, 255, 1);
        background-color: transparent;
        height: 30px;
        line-height: 30px;;
        font-size: 14px;
        border: 1px solid rgba(255, 255, 255, 1);
        padding: 5px 10px;
        border-radius: 30px;
      }
    }

    .background {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-size: cover;
      z-index: 0;
      border-radius: 5px;

      .fade {
        width: 100%;
        height: 100%;
        background: linear-gradient(0deg, #2A4059 -6.16%, rgba(42, 64, 89, 0.23) 109.19%);
        border-radius: 5px;
      }
    }
  }

  .createGroup > div {
    width: 50%;
    height: 300px;
    display: flex;
    justify-content: left;
    align-items: center;
    padding-left: 15px;
    width: 160px;

    a {
      border: 1px solid rgba(192, 197, 205, 1.00);
      height: 30px;
      background-color: transparent;
      font-size: 14px;
      color: rgba(128, 140, 156, 1.00);
      line-height: 30px;
      padding: 0 15px;
      border-radius: 30px;

      &:hover {
        text-decoration: none;
      }
    }
  }
}

.groups :global(.slick-slide > div) {
  padding: 0 8px 15px 8px;
}

.groups :global(.slick-list) {
  padding: 0 20% 0 0;
}

@media screen and (max-width: 600px) {
  .groups {
    padding: 0 0 0 0;
  }
}
