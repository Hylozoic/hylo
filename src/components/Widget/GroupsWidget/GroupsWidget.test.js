import React from 'react'
import { shallow } from 'enzyme'
import GroupsWidget from './GroupsWidget'

const defaultMinProps = {
  routeParams: { context: 'groups', slug: 'group one'},
  items: []
}

function renderComponent (renderFunc, props = {}) {
  return renderFunc(
    <GroupsWidget {...{ ...defaultMinProps, ...props }} />
  )
}

describe('GroupsWidget', () => {
  it('renders correctly (with min props)', () => {
    const wrapper = renderComponent(shallow)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with items', () => {
    const props = {
      items: [
        { id: 1, slug: 'slug2', name: 'group one', avatarUrl: 'https://google.com', description: 'yo', memberCount: 1 },
        { id: 2, slug: 'slug2', name: 'group 2', avatarUrl: 'https://google.com', description: 'oy', memberCount: 10 }
      ]
    }
    const wrapper = renderComponent(shallow, props)
    expect(wrapper).toMatchSnapshot()
  })

})
