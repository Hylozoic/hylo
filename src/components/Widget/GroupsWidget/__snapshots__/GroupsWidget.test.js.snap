// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupsWidget renders correctly (with min props) 1`] = `
<div
  data-stylename="groups"
>
  <Slider
    arrows={true}
    dots={true}
    infinite={false}
    responsive={
      Array [
        Object {
          "breakpoint": 600,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 1.1,
          },
        },
      ]
    }
    slidesToScroll={1}
    slidesToShow={2.1}
    speed={500}
  >
    <div
      data-stylename="createGroup"
    >
      <div>
        <Link
          to="/all/create/group"
        >
          + Create Group
        </Link>
      </div>
    </div>
  </Slider>
  <div
    data-stylename="groupBumper"
  />
</div>
`;

exports[`GroupsWidget renders correctly with items 1`] = `
<div
  data-stylename="groups"
>
  <Slider
    arrows={true}
    dots={true}
    infinite={false}
    responsive={
      Array [
        Object {
          "breakpoint": 600,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 1.1,
          },
        },
      ]
    }
    slidesToScroll={1}
    slidesToShow={2.1}
    speed={500}
  >
    <GroupCard
      group={
        Object {
          "avatarUrl": "https://google.com",
          "description": "yo",
          "id": 1,
          "memberCount": 1,
          "name": "group one",
          "slug": "slug2",
        }
      }
      key="1"
      routeParams={
        Object {
          "context": "groups",
          "slug": "group one",
        }
      }
    />
    <GroupCard
      group={
        Object {
          "avatarUrl": "https://google.com",
          "description": "oy",
          "id": 2,
          "memberCount": 10,
          "name": "group 2",
          "slug": "slug2",
        }
      }
      key="2"
      routeParams={
        Object {
          "context": "groups",
          "slug": "group one",
        }
      }
    />
    <div
      data-stylename="createGroup"
    >
      <div>
        <Link
          to="/all/create/group"
        >
          + Create Group
        </Link>
      </div>
    </div>
  </Slider>
  <div
    data-stylename="groupBumper"
  />
</div>
`;
